docs/architecture.md

```diff
@@ -0,0 +1,23 @@
+# Packages
+
+All packages follow Gatsby's [plugin naming scheme](https://www.gatsbyjs.com/docs/plugins-themes-and-starters/). 
+
+* **gatsby-plugin-graphql** adds Webpack and Babel configuration so we can share queries and types between Gatsby and any other GraphQL server. This plugin is essential for dynamic, client side queries
+
+* **gatsby-plugin-i18n** is an optimized react-intl capable plugin. This plugin pre-optimizes messages we can serve react-intl without an ICU parser to the browser. Also, this plugins adds all necessary providers to the React tree
+
+* **gatsby-plugin-theme-ui** is an optimized theme-ui plugin for Gatsby. It includes many optmizations, not only on `theme-ui` but also on `emotion`. The goal is to merge this package into their community based versions so all the community benefits from such optimizations
+
+* **gatsby-plugin-nginx** allows you to serve your Gatsby site using `nginx`. This plugin creates a Netlify like experience where all dynamic pages, redirects and reverse proxies are merged automatically into a `nginx.conf` file. Just pick this file and start serving your site from anywhere
+
+* **gatsby-source-vtex** connects the Gatsby data layer to VTEX's GraphQL graph
+
+* **gatsby-theme-store** contains a Web SDK and is the main skeleton of an ecommerce. In it you will find the web's best practices, like above/below the fold techniques an first class Pixel integration. Note that this repo should NOT contain any non structurant component
```

Comment:
Not clear what is the API here. Maybe small pieces would be easier to maintain and do breaking changes.

Reply:
This package should contain basically `/sdk` `/pages` `/templates` and some structuring components in `/components`, like BelowTheFold.tsx and AboveTheFold.tsx from `gatsby-theme-vtex`

The `/sdk` should contain all basic store logic, like search pagination, sales channel, auth, queries, etc.
The `/pages` and `/templates` are the webpack entrypoints
The `/components` have  components that MUST be shadowed, like Layout, BelowTheFold.tsx and AboveTheFold.tsx

Reply:
Does it make sense to split the sdk packages? I'm imagining a situation where we want to do a breaking change of a specific SDK, the user wouldn't need to update the basic plugin.

Also **gatsby-theme-vtex-store** isn't a better name?

Reply:
I'm really not sure about this one. The tradeoff I see in here is that splitting the packages give you greater flexibility, however maintaining a monolithic structure is easier for new people joining and makes us keep the codebase small. However, I really don't know.

About the package naming scheme, I think these packages can be the beginning of the OpenStore initiative